sov-modules/sov-modules-impl/integration-tests/tests/demo/basic_runtime.rs

```diff
@@ -0,0 +1,199 @@
+//! Basic Runtime Example
+//! This `Runtime` serves as a basic example of how to wire up module system and trigger the rollup logic.
+
+#![allow(dead_code)]
+
+use sov_modules_api::{
+    mocks::{MockContext, MockPublicKey},
+    CallResponse, Context, DispatchCall, DispatchQuery, Error, Genesis, Module,
+};
+use sov_modules_macros::{DispatchCall, DispatchQuery, Genesis, MessageCodec};
+use sov_state::{CacheLog, ValueReader};
+use sovereign_db::state_db::StateDB;
+
+/// dispatch_tx is a high level interface used by the sdk.
+/// Transaction signature must be checked outside of this function.
+fn dispatch_tx<C: Context, VR: ValueReader>(
+    _tx_data: Vec<u8>,
+    _context: C,
+    _value_reader: VR,
+) -> Result<(CallResponse, CacheLog), Error> {
+    // 1. Create Storage (with fresh Cache)
```

Comment:
(non-blocking): In the zk enviornment, we might want to pass in the storage from the context. That way, if the prover knows that a transaction won't revert they can run it against the main storage and save all the overhead of creating and merging an extra cache.

But that's an optimization, and it only applies in zk - so definitely not a blocker. Maybe not worth doing at all.

Reply:
This is a good point. Let's discuss it tmr on the daily.